app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -51,6 +55,29 @@ export const snapKeyringBuilder = (
         const addresses = await getKeyringController().getAccounts();
         return addresses.includes(address.toLowerCase());
       },
+      redirectUser: async (snapId: string, url: string, message: string) => {
+        // Either url or message must be defined
+        if (url !== '' || message !== '') {
```

Comment:
Could we check the length instead?

Also in what case would these not be defined? Do you not need to specify a URL to redirect? 🤔

Reply:
@FrederikBolding the url and message come from the [eth-snap-keyring](https://github.com/MetaMask/eth-snap-keyring/pull/136) which is sent from a snap ([example](https://github.com/MetaMask/snap-simple-keyring/pull/98/files#diff-22aa27652c8dffcd76652c4ed22b9e646ab8cc3b43594d0eaa303e272017a367R219-R221)) so they can never be undefined but they can be empty strings. we can handle the case where one of them is not set but when both are empty we wont do anything. See the [designs ](https://www.figma.com/file/K0csZegZ5XQ6TjfKk3n4Z8/Add-dialog-to-inform-about-redirection-to-snap-site-%2335?type=design&node-id=604-2586&mode=design&t=tRpxcs4qeRiRKxPl-0)

Reply:
are they asserted somewhere to guarantee that they are strings? If so, I think checking the length is slightly cleaner than checking against `''`

Reply:
we convert all the values to '' if they are null [here](https://github.com/MetaMask/eth-snap-keyring/pull/136/files#diff-6c3541c9e03e286cae318f6482c33d1a808662e60c02c91ea5dea83b28e91bb1R265). That being said I added the length check in the latest commit.